File: elixir/acronym/acronym.exs
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    words = string
            |> Macro.underscore               # convert camel case to snake case
            |> String.replace(~r/[_-]/, " ")  # replace - & _ with spaces
            |> String.split                   # split on spaves
    # IO.puts "\nWORDS"
    # Enum.map(words, fn e -> IO.puts e end)

    acro  = words
            |> Enum.map_join( &String.first(&1) ) # get first letter of each word & join
            |> String.upcase                      # upcase string
    # IO.puts "-----\nANS: #{acro}"
    acro
  end
end

File: ruby/acronym/acronym.rb
class Acronym

  FIRST_LETTER_REGEX  = { en: /\b[a-zA-Z]/,
                          de: /\b[äöüßa-zÄÖÜA-Z]/,
                          it: /\b[áéíóúa-zÁÉÍÓÚA-Z]/,
                        }
  def self.abbreviate multiple_words, lang = :en
    acronym = multiple_words.scan(FIRST_LETTER_REGEX[lang]).join.upcase
    raise ArgumentError, "An acronym requires multiple words" unless acronym.length > 1

    acronym
  end

end
